--- a/cpp/include/IceUtil/Optional.h
+++ b/cpp/include/IceUtil/Optional.h
@@ -45,6 +45,14 @@
     Optional(IceUtilInternal::NoneType) : _isSet(false)
     {
     }
+    
+    /**
+     * Constructs an optional as a copy of the given optional.
+     * @param r The source optional.
+     */
+    Optional(const Optional& r) : _value(r._value), _isSet(r._isSet)
+    {
+    }
 
     /**
      * Constructs an optional with the given value.
