--- a/cpp/config/Make.rules.Linux
+++ b/cpp/config/Make.rules.Linux
@@ -162,7 +162,7 @@
 # QT is used only for the deprecated IceGrid and IceStorm SQL plugins
 #
 ifneq ($(QT_HOME),)
-   QT_FLAGS             = -I$(QT_HOME)/include
-   QT_LIBS              = -L$(QT_HOME)/$(libsubdir) -lQtCore -lQtSql
+   QT_FLAGS             = -I$(QT_HOME)
+   QT_LIBS              = -lQtCore -lQtSql
    QT_RPATH_LINK        = $(call rpathlink,$(QT_HOME)/$(libsubdir))
 endif
\ No newline at end of file
