--- a/cpp/config/Make.rules.Linux
+++ b/cpp/config/Make.rules.Linux
@@ -45,12 +45,12 @@
       #
       # We are an ultra, at least, and so have the atomic instructions
       #
-      $(warning ===================================================================) 
+      $(warning ===================================================================)
       $(warning Linux on SPARC is currently unsupported. The Ice team does not)
       $(warning maintain SPARC specific portions of the source code or build)
       $(warning system. Contact sales@zeroc.com if you wish to sponsor official)
       $(warning support.)
-      $(warning ===================================================================) 
+      $(warning ===================================================================)
       USE_SPARC_ASM	= yes
       CXXARCHFLAGS	+= -mcpu=ultrasparc -pipe -Wno-deprecated -DUSE_SPARC_ASM
    endif
@@ -61,12 +61,12 @@
       # On sun4m, there is a bug in some CPU/kernel/gcc configurations which
       # prevent us from using '-mcpu=v8'
       #
-      $(warning ===================================================================) 
+      $(warning ===================================================================)
       $(warning Linux on SPARC is currently unsupported. The Ice team does not)
       $(warning maintain SPARC specific portions of the source code or build)
       $(warning system. Contact sales@zeroc.com if you wish to sponsor official)
       $(warning support.)
-      $(warning ===================================================================) 
+      $(warning ===================================================================)
       USE_SPARC_ASM	= no
       CXXARCHFLAGS	+= -mtune=v8 -pipe -Wno-deprecated -DICE_USE_MUTEX_SHARED
    endif
@@ -86,9 +86,9 @@
    endif
 
    ifeq ($(OPTIMIZE),yes)
-      CXXFLAGS		+= -O2 -DNDEBUG
+      CXXFLAGS		+= -g -O2 -DNDEBUG
    else
-      CXXFLAGS		+= -g 
+      CXXFLAGS		+= -g
    endif
 
    #
@@ -100,10 +100,10 @@
 
    mklib		= ar cr $(1) $(2)
 
-   rpathlink            = -Wl,-rpath-link,$(1) 
+   rpathlink            = -Wl,-rpath-link,$(1)
 
    ifneq ($(embedded_runpath_prefix),)
-      LDPLATFORMFLAGS      = -Wl,--enable-new-dtags -Wl,-rpath,$(runpath_libdir)
+      LDPLATFORMFLAGS      = -Wl,--enable-new-dtags
    else
       LDPLATFORMFLAGS      = -Wl,--enable-new-dtags
    endif
@@ -113,20 +113,20 @@
 endif
 
 ifeq ($(CXX),icpc)
-   $(warning ===================================================================) 
+   $(warning ===================================================================)
    $(warning Intel C++ is currently not supported. The Ice team does not)
    $(warning maintain Intel C++ specific portions of the source code or build)
    $(warning system.  Contact sales@zeroc.com if you wish to sponsor official)
    $(warning support.)
-   $(warning ===================================================================) 
-   CXXFLAGS		= -D_REENTRANT    
+   $(warning ===================================================================)
+   CXXFLAGS		= -D_REENTRANT
 
    ifneq ($(GENPIC),no)
       CXXFLAGS		+= -fPIC
    endif
 
    ifeq ($(OPTIMIZE),yes)
-      CXXFLAGS		+= -O2 -DNDEBUG
+      CXXFLAGS		+= -g -O2 -DNDEBUG
    else
       CXXFLAGS		+= -g
    endif
@@ -140,7 +140,7 @@
 
    mklib		= ar cr $(1) $(2)
 
-   rpathlink            = -Wl,-rpath-link,$(1) 
+   rpathlink            = -Wl,-rpath-link,$(1)
 
 endif
 
@@ -165,4 +165,4 @@
    QT_FLAGS             = -I$(QT_HOME)
    QT_LIBS              = -lQtCore -lQtSql
    QT_RPATH_LINK        = $(call rpathlink,$(QT_HOME)/$(libsubdir))
-endif
\ No newline at end of file
+endif
