Index: ice-debian-packaging/config/Make.rules.Linux
===================================================================
--- ice-debian-packaging.orig/config/Make.rules.Linux	2018-04-24 11:29:50.727659336 +0000
+++ ice-debian-packaging/config/Make.rules.Linux	2018-10-26 15:54:29.902521047 +0000
@@ -7,7 +7,9 @@
 #
 # **********************************************************************
 
-linux_id                        = $(if $(wildcard /etc/os-release),$(shell . /etc/os-release && echo $${ID}),)
+known_distributions             := centos rhel fedora debian ubuntu amzn sles
+linux_ids                       := $(if $(wildcard /etc/os-release),$(shell . /etc/os-release && echo $${ID} $${ID_LIKE}),)
+linux_id                        := $(word 1,$(or $(filter $(known_distributions),$(linux_ids)),$(linux_ids)))
 is-bin-program                  = $(and $(filter $(bindir)%,$($4_targetdir)),$(filter $($1_target),program))
 
 ifneq ($(OECORE_SDK_VERSION),)
