Index: ice-debian-packaging/cpp/config/Make.rules
===================================================================
--- ice-debian-packaging.orig/cpp/config/Make.rules
+++ ice-debian-packaging/cpp/config/Make.rules
@@ -98,7 +98,7 @@ endif
 # directory.
 #
 cpp11_cppflags          = -DICE_CPP11_MAPPING
-ifneq ("OK", $(shell $(CC) -E config/cplusplus_check.cpp 2> /dev/null && echo "OK"))
+ifneq ($(shell $(CC) -E config/cplusplus_check.cpp 1> /dev/null 2>&1; echo $$?),0)
 cpp11_cppflags          := $(cpp11_cppflags) -std=c++11
 endif
 cpp11_ldflags           = $(cpp11_cppflags)
Index: ice-debian-packaging/cpp/test/Ice/ami/AllTests.cpp
===================================================================
--- ice-debian-packaging.orig/cpp/test/Ice/ami/AllTests.cpp
+++ ice-debian-packaging/cpp/test/Ice/ami/AllTests.cpp
@@ -7,6 +7,10 @@
 #include <TestHelper.h>
 #include <Test.h>
 
+#ifdef ICE_CPP11_MAPPING
+#   include <thread>
+#endif
+
 using namespace std;
 
 namespace
