--- a/java/src/IceGridGUI/plain-jar.gradle
+++ b/java/src/IceGridGUI/plain-jar.gradle
@@ -20,7 +20,7 @@
             delegate.manifest {
                 attribute(name: "Main-Class", value: "com.zeroc.IceGridGUI.Main")
                 attribute(name: "Built-By", value: "ZeroC, Inc.")
-                attribute(name: "Class-Path", value: configurations.runtime.resolve().collect { it.toURI() }.join(' '))
+                attribute(name: "Class-Path", value: configurations.runtime.resolve().collect { "file://${it.toURI().getRawPath()}" }.join(' '))
             }
         }
     }
@@ -48,7 +48,7 @@
                 attribute(name: "Main-Class", value: "com.zeroc.IceGridGUI.Main")
                 attribute(name: "Built-By", value: "ZeroC, Inc.")
                 attribute(name: "Class-Path", value: configurations.runtime.resolve().collect {
-                    it.toURI() }.join(' ').replaceAll("${libDir.replaceAll('\\\\', '/')}", "${jarDir.replaceAll('\\\\', '/')}"))
+                    "file://${it.toURI().getRawPath()}" }.join(' ').replaceAll("${libDir.replaceAll('\\\\', '/')}", "${jarDir.replaceAll('\\\\', '/')}"))
             }
         }
     }
