Index: ice-debian-packaging/python/modules/IcePy/Util.cpp
===================================================================
--- ice-debian-packaging.orig/python/modules/IcePy/Util.cpp	2018-04-24 11:29:51.059654290 +0000
+++ ice-debian-packaging/python/modules/IcePy/Util.cpp	2018-10-17 20:42:03.687623802 +0000
@@ -73,6 +73,11 @@
         }
         v.major = static_cast<Ice::Byte>(m);
     }
+    else
+    {
+        v.major = 0;
+    }
+
     if(minor.get())
     {
         major = PyNumber_Long(minor.get());
@@ -89,6 +94,10 @@
         }
         v.minor = static_cast<Ice::Byte>(m);
     }
+    else
+    {
+        v.minor = 0;
+    }
     return true;
 }
 
