Index: ice-debian-packaging/cpp/src/Ice/NetworkProxy.cpp
===================================================================
--- ice-debian-packaging.orig/cpp/src/Ice/NetworkProxy.cpp
+++ ice-debian-packaging/cpp/src/Ice/NetworkProxy.cpp
@@ -81,7 +81,6 @@ private:
 SOCKSNetworkProxy::SOCKSNetworkProxy(const string& host, int port) : _host(host), _port(port)
 {
     assert(!host.empty());
-    memset(&_address, 0, sizeof(_address));
 }
 
 SOCKSNetworkProxy::SOCKSNetworkProxy(const Address& addr) : _port(0), _address(addr)
@@ -194,7 +193,6 @@ HTTPNetworkProxy::HTTPNetworkProxy(const
     _host(host), _port(port), _protocol(EnableBoth)
 {
     assert(!host.empty());
-    memset(&_address, 0, sizeof(_address));
 }
 
 HTTPNetworkProxy::HTTPNetworkProxy(const Address& addr, ProtocolSupport protocol) :
Index: ice-debian-packaging/cpp/src/IceGrid/Database.cpp
===================================================================
--- ice-debian-packaging.orig/cpp/src/IceGrid/Database.cpp
+++ ice-debian-packaging/cpp/src/IceGrid/Database.cpp
@@ -1628,7 +1628,7 @@ Database::updateObject(const Ice::Object
 
             if(!_objects.get(txn, id, info))
             {
-                ObjectNotRegisteredException(id);
+                throw ObjectNotRegisteredException(id);
             }
             info.proxy = proxy;
             addObject(txn, info, false);
