--- a/java/src/IceGridGUI/src/main/java/IceGridGUI/Main.java
+++ b/java/src/IceGridGUI/src/main/java/IceGridGUI/Main.java
@@ -39,7 +39,10 @@ public class Main extends JFrame
             }
             else  // JGoodies L&F
             {
-                UIManager.setLookAndFeel("com.jgoodies.looks.plastic.PlasticLookAndFeel");
+                //
+                // Setting PlasticLookAndFeel trigger a crash when creating a JFileChooser.
+                //
+                //UIManager.setLookAndFeel("com.jgoodies.looks.plastic.PlasticLookAndFeel");
             }
         }
         catch(Exception e)
