--- a/java/gradle/library.gradle
+++ b/java/gradle/library.gradle
@@ -136,16 +136,20 @@
         into "${DESTDIR}${jarDir}"
     }
 
-    task updateInstallManifest(dependsOn: copyJars) << {
-        ant.jar(update: true, destfile: "${DESTDIR}${jarDir}/${jar.archiveName}") {
-            delegate.manifest {
-                attribute(name: "Built-By", value: "ZeroC, Inc.")
-                attribute(name: "Class-Path", value: configurations.runtime.resolve().collect { "file://${it.absolutePath}" }.join(' ').replaceAll("${libDir}", "${jarDir}"))
+    task updateInstallManifest(dependsOn: copyJars) {
+        outputs.file("${DESTDIR}${jarDir}/${jar.archiveName}")
+        doLast {
+            ant.jar(update: true, destfile: "${DESTDIR}${jarDir}/${jar.archiveName}") {
+                delegate.manifest {
+                    attribute(name: "Built-By", value: "ZeroC, Inc.")
+                    attribute(name: "Class-Path", value: configurations.runtime.resolve().collect { "file://${it.absolutePath}" }.join(' ').replaceAll("${libDir}", "${jarDir}"))
+                }
             }
         }
     }
-    updateInstallManifest.files("${DESTDIR}${jarDir}/${jar.archiveName}")
-    install.dependsOn(updateInstallManifest)
+
+    task install(dependsOn: updateInstallManifest, overwrite:true) {
+    }
 } else {
     task install(type: Copy, dependsOn: jarSources, overwrite: true) {
         from "${pomName}"
