Fix ftbfs with python3.9
--- a/python/modules/IcePy/Init.cpp
+++ b/python/modules/IcePy/Init.cpp
@@ -141,10 +141,13 @@
     Ice::registerIceDiscovery(false);
     Ice::registerIceLocatorDiscovery(false);
 
+#if PY_VERSION_HEX < 0x03090000
     //
-    // Notify Python that we are a multi-threaded extension.
+    // Notify Python that we are a multi-threaded extension. This method is deprecated
+    // and does nothing since Python 3.9
     //
     PyEval_InitThreads();
+#endif
 
 #if PY_VERSION_HEX >= 0x03000000
     //
